plutus-core/plutus-ir/test/transform/inline/letOverAppType3.golden

```diff
@@ -3,5 +3,11 @@
   (termbind
     (strict) (vardecl idFun (all a (type) (fun a a))) (abs a (type) (lam x a x))
   )
-  (con integer 3)
+  [
+    (lam x (con integer) x)
+    [
+      [ (lam x (fun (con integer) (con integer)) x) (lam x (con integer) x) ]
+      (con integer 3)
+    ]
+  ]
```

Comment:
Actually, I think these examples are a much clearer example of something weird happening. I think what's happened is that we've decided to inline `const`... but as a 0-arity term, so we don't do any beta-reduction! And that's pretty weird! Surely it would be even smaller if we did the beta-reduction? So why are we not taking that path?